checks.d/sidekiq.py

```diff
@@ -0,0 +1,246 @@
+'''
+Sidekiq checks
+'''
+# stdlib
+import time
+
+# 3rd party
+import redis
+
+# project
+from checks import AgentCheck
+
+class Stats:
+    def __init__(self, conn, namespace):
+        self.conn = conn
+        self.namespace = namespace
+        self._script = self._register_script()
+        self._check_stats()
+
+    # This script collects all the sidekiq related stats in one go.
+    # KEYS[1] = 'stat:processed'
+    # KEYS[2] = 'stat:failed'
+    # KEYS[3] = 'schedule'
+    # KEYS[4] = 'retry'
+    # KEYS[5] = 'dead'
+    # KEYS[6] = 'processes'
+    # KEYS[7] = 'queues'
+    #
+    # Returns an array of results. The array follows the following format:
+    # [
+    #   processed,
+    #   failed,
+    #   scheduled_size,
+    #   retry_size,
+    #   dead_size,
+    #   processes_size,
+    #   workers_size,
+    #   number_of_queues,
+    #   queue name 1,
+    #   queue name 2,
+    #   ...,
+    #   queue name n,
+    #   queue length 1,
+    #   queue length 2,
+    #   ...,
+    #   queue length n,
+    #   queue latency 1,
+    #   queue latency 2,
+    #   ...,
+    #   queue latency n
+    # ]
+    #
+    def _register_script(self):
+        s = """
+        local processed = redis.call("GET", KEYS[1])
+        local failed = redis.call("GET", KEYS[2])
+        local scheduled_size = redis.call("ZCARD", KEYS[3])
+        local retry_size = redis.call("ZCARD", KEYS[4])
+        local dead_size = redis.call("ZCARD", KEYS[5])
+        local processes_size = redis.call("SCARD", KEYS[6])
+        local processes = redis.call("SMEMBERS", KEYS[6])
+
+        local workers_size = 0
+        for i, pkey in ipairs(processes) do
+            local nspkey = ARGV[1] == "None" and pkey or (ARGV[1] .. ":" .. pkey)
+            workers_size = workers_size + tonumber(redis.call("HGET", nspkey, "busy"))
```

Comment:
I've notice that script fails on this line time to time, with `user_script:13: attempt to perform arithmetic on a nil value`. This is because `tonumber` on `nil` returns `nil`. After changing this to `(tonumber(redis.call("HGET", nspkey, "busy")) or 0)` fails stopped.